src/wp-includes/blocks/template-part.php

```diff
@@ -249,4 +249,8 @@ function register_block_core_template_part() {
 		)
 	);
 }
-add_action( 'init', 'register_block_core_template_part' );
+/*
+ * This block relies on generated styles from previously registered blocks, so
+ * it should be registered last.
+ */
+add_action( 'init', 'register_block_core_template_part', 29 );
```

Comment:
Note that we'd need to make this change to the corresponding file in Gutenberg, since Core's build process uses the files from the `@wordpress/block-library` package as its source of truth for dynamic blocks' PHP code and overwrites any changes we make here 😬 

(I think that's also the reason for the test failures we're seeing in CI.)